xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
> I assume on X/intel you could catch up because there are more buffers on the client side?

well, two buffers are enough. I would have thought by requesting a visual with double buffers we get two buffers. On X11 you can clearly see that we get a frame ahead and if we are not a frame ahead, there is no blocking.

> Exactly, that's why I asked: Take the time before or after swapBuffers?

We need the time from last vblank. If swapbuffers blocks, then after. On X11 it is not that trivial, because swapBuffers does not necessarily block (see above and X11 on my branch)